sdks/java/core/src/test/java/org/apache/beam/sdk/options/PipelineOptionsFactoryTest.java

```diff
@@ -809,6 +849,57 @@ public void testSetASingularAttributeUsingAListIsIgnoredWithoutStrictParsing() {
     expectedLogs.verifyWarn("Strict parsing is disabled, ignoring option");
   }
 
+  /** A test interface containing all supported List return types. */
+  public static interface Maps extends PipelineOptions {
+    Map<Integer, Integer> getMap();
+    void setMap(Map<Integer, Integer> value);
+
+    Map<Integer, Map<Integer, Integer>> getNestedMap();
+    void setNestedMap(Map<Integer, Map<Integer, Integer>> value);
+  }
+
+  @Test
+  public void testMapIntInt() {
+    String[] manyArgsShort =
+        new String[] {"--map={\"1\":1,\"2\":2}"};
+    String[] oneArg = new String[] {"--map={\"1\":1}"};
+    String[] missingArg = new String[] {"--map="};
+
+    Maps options = PipelineOptionsFactory.fromArgs(manyArgsShort).as(Maps.class);
+    assertEquals(ImmutableMap.of(1, 1, 2, 2), options.getMap());
+    options = PipelineOptionsFactory.fromArgs(oneArg).as(Maps.class);
+    assertEquals(ImmutableMap.of(1, 1), options.getMap());
+
+    expectedException.expect(IllegalArgumentException.class);
+    expectedException.expectMessage(
+      "Empty argument value is only allowed for String, String Array, and "
+      + "Collection, but received: java.util.Map<java.lang.Integer, java.lang.Integer>");
```

Comment:
I thought above I saw you reworded this to "Collection of Strings" -- is this no longer true?